### ISSUE TITLE
Guitar mode implementation

### ISSUE THREAD

**Issue by klaviotarka** [ppy/osu#12611]
**Describe the new feature:** 
   Just like any mode in Osu! this will be a new mode called Guitar or anything fancier. The point of this mode would be specifically for guitar players or people who want to get better at guitar generally.  You would get a screen similar to manias screen and play along the song while strumming the guitar strings.

**Proposal designs of the feature:** 

First I'd like to start out by saying, I started this request as a question if implementing this mode would be ok with the game terms & rules and figured this would be the best place to ask peppy about it.

Secondly I don't expect to get help developing this mode and started it entirely as a way to practice both a new programming language and hopefully I can make the mode a reality so everyone can get a chance to improve much faster at guitar (struggling with this myself because I cant find many songs, which wont be a problem if there's an entire community supporting the mode).

**Project idea so far**
The guitar cable will need to be connected to an adapter and the latter is connected to the target PC. Thus the PC will be able to receive guitar noise in real time, so no delay.

I believe there will be 3 key topics that need to be solved before the mode can be successfully implemented 

**1. Converting guitar audio into readable code.
2. Figuring out how to turn the converted code into variables that can be implemented on simple integers.
3. Implementing the code in Osu!**

**Tl;dr:** Wondering if implementing a guitar mode where people can play with a real guitar would be acceptable.

**Comment by peppy** [comment]
Rulesets in lazer can be made by anyone (take a look at https://github.com/ppy/osu/issues/5852 for examples, and check the templates we provide to get started https://github.com/ppy/osu/tree/master/Templates)

That said, we don't keep issues open for proposals for new rulesets. Good luck with your project - feel free to reach out on discord if you need help with osu!/framework.

**Comment by MasicoreLord** [comment]
Ah so like Rocksmith, but in osu!
